### PR TITLE
test: replace nosetest with pytest

### PR DESCRIPTION
**Pull request by aabadie**
This PR switch the testing tool from nosetests (not maintained anymore) to pytest.

It works when run locally but it needs further testing (not done yet) with integration tests.

### CODE REVIEW

**Comment by aabadie** [comment]
This PR is based on #20

**Comment by rienafairefr** [comment]
Great !

**Comment by rienafairefr** [comment]
Are you still working on the host user right thing ? I feel like this pytest PR could be decoupled from the host user/gosu ? I can do it if you need

**Comment by aabadie** [comment]
> I feel like this pytest PR could be decoupled from the host user/gosu ? I can do it if you need

Yes that would be great. Regarding the host user (see #20), I have problems that might be worth discussing (next week). The branch there is not up-to-date

**Comment by rienafairefr** [comment]
I can push to your fork to update that PR, pytest seems to work in local. like you want.

**Comment by aabadie** [comment]
> I can push to your fork to update that PR

Yes go ahead, thanks !

**Comment by rienafairefr** [comment]
done, just need to verify that codecov gets the uploaded results with the pytest-generated coverage xml

**Comment by aabadie** [comment]
Codecov seems to work

**Comment by aabadie** [comment]
pytest 3.3 output is not as nice as locally

**Comment by aabadie** [comment]
@rienafairefr, I spent some time making the switch to pytest fully work with integration tests (on real hardware). The last commit does this. 
Regarding coverage, there's a problem that is already in master. No report have been pushed for a moment. Will open a PR for this.

If code changes are ok for you, this PR can be merged.

File: tox.ini [hunk 1]
```diff
@@ -32,37 +38,37 @@ whitelist_externals = /bin/bash
...
 whitelist_externals =
     /bin/bash
-    /bin/hostname
-# Use only sitepackages on gateways, no 'deps'
-# Only install 'webtest' as I'm lazy to package it for the moment
-deps = WebTest
-# Requires tox >= 1.8
-sitepackages = True
+    /usr/bin/make
+deps = -rtests_utils/test-requirements.txt
```
Comment (rienafairefr) [inline tox.ini:48]: the WebTest dependency is not used for anything in the end ?

Comment (aabadie) [reply tox.ini:48]: It's already installed by the `test-requirements.txt`. The thing is that before it was using the one that is packaged by yocto. I think it was to be as close as possible as a real environment.
When using the test-requirements.txt file, testing dependencies are installed in the tox env. It's simpler this way I think but maybe there are other things I missing.

**Comment by rienafairefr** [comment]
Looks good, let's merge :+1:

**Comment by aabadie** [comment]
I'm running a remote integration test on the m3 in devgrenoble. It works well.

**Pull request merged**